src/main/java/org/sasanlabs/service/vulnerability/fileupload/UnrestrictedFileUpload.java

```diff
@@ -357,6 +365,7 @@ Path getContentDispositionRoot() {
             description = "UNRESTRICTED_FILE_UPLOAD_NO_VALIDATION_FILE_NAME")
     @VulnerableAppRequestMapping(
             value = LevelConstants.LEVEL_8,
+            variant = Variant.SECURE,
```

Comment:
it is not secure, it causes path traversaal attack. try naming file as "../karan.png", it will upload to parent directory

Reply:
Yes, I cannot find the target directory (in my case it was not created in my destination folder), so I will add only a proper description.

Reply:
hmm yeah there is a bug in this. can you please use docker for exploiting this vulnerability?

Reply:
https://hub.docker.com/r/sasanlabs/owasp-vulnerableapp